tests/route/test_static_route.py

```diff
@@ -0,0 +1,108 @@
+            tcp_dport=4321)
+
+    exp_pkt = pkt.copy()
+    exp_pkt = mask.Mask(exp_pkt)
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+    if ipv6:
+        exp_pkt.set_do_not_care_scapy(packet.IPv6, 'hlim')
+        exp_pkt.set_do_not_care_scapy(packet.IPv6, 'chksum')
+    else:
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'ttl')
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'chksum')
+
+    testutils.send(ptfadapter, 5, pkt)
+    testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=expected_ports)
+
+def run_static_route_test(duthost, ptfadapter, ptfhost, prefix, nexthop_addrs, prefix_len, nexthop_devs, ipv6=False):
+    # Add ipaddresses in ptf
+    add_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=ipv6)
+
+    try:
+        # Add static route
+        duthost.shell("sonic-db-cli CONFIG_DB hmset 'STATIC_ROUTE|{}' nexthop {}".format(prefix, ",".join(nexthop_addrs)))
+        time.sleep(2)
+
+        # Check traffic get forwarded to the nexthop
+        ip_dst = str(ipaddress.ip_network(unicode(prefix))[1])
+        generate_and_verify_traffic(duthost, ptfadapter, ip_dst, nexthop_devs, ipv6=ipv6)
+
+    finally:
+        # Remove static route
+        duthost.shell("sonic-db-cli CONFIG_DB del 'STATIC_ROUTE|{}'".format(prefix), module_ignore_errors=True)
+
+        # Delete ipaddresses in ptf
+        del_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=ipv6)
+
+def get_vlan_info(duthost, tbinfo, ipv6=False):
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    vlan_intf = mg_facts['minigraph_vlan_interfaces'][1 if ipv6 else 0]
+    prefix_len = vlan_intf['prefixlen']
+    vlan_subnet = ipaddress.ip_network(vlan_intf['subnet'])
+    return prefix_len, vlan_subnet
+
+def test_static_route(duthost, ptfadapter, ptfhost, tbinfo):
+    prefix_len, vlan_subnet = get_vlan_info(duthost, tbinfo)
+    run_static_route_test(duthost, ptfadapter, ptfhost, "1.1.1.0/24", 
+        [str(vlan_subnet[11])], prefix_len, [1])
+
+def test_static_route_ecmp(duthost, ptfadapter, ptfhost, tbinfo):
+    prefix_len, vlan_subnet = get_vlan_info(duthost, tbinfo)
+    run_static_route_test(duthost, ptfadapter, ptfhost, "2.2.2.0/24", 
+        [str(vlan_subnet[21]), str(vlan_subnet[22]), str(vlan_subnet[23])], prefix_len, [1, 2, 3])
+
+def test_static_route_ipv6(duthost, ptfadapter, ptfhost, tbinfo):
+    prefix_len, vlan_subnet = get_vlan_info(duthost, tbinfo, ipv6=True)
+    run_static_route_test(duthost, ptfadapter, ptfhost, "2000:1::/64", 
+        [str(vlan_subnet[11])], prefix_len, [1], ipv6=True)
+
+def test_static_route_ecmp_ipv6(duthost, ptfadapter, ptfhost, tbinfo):
+    prefix_len, vlan_subnet = get_vlan_info(duthost, tbinfo, ipv6=True)
+    run_static_route_test(duthost, ptfadapter, ptfhost, "2000:2::/64", 
+        [str(vlan_subnet[21]), str(vlan_subnet[22]), str(vlan_subnet[23])], prefix_len, [1, 2, 3], ipv6=True)
```

Comment:
I think you are using eth0, eth1 and eth2 of ptf interface to configure IP. For some t0 testbeds, this may be corresponding to a portchannel interface or a port that is down. I think a better option is to get the ptf interfaces for the vlan ports and use it here.

Reply:
Using ptf interfaces for vlan ports instead.